webrtc/videoroom/assets/src/membraneWebRTC.ts

```diff
@@ -101,12 +101,31 @@ export class MembraneWebRTC {
     this.connection = undefined;
   };
 
+  public muteMicrophone = async () => {
+    this.localTracks.forEach(([track, stream]) => {
+      if (track.kind == "audio") {
+        track.enabled = !track.enabled;
+        this.channel.push("mute-microphone", { data: !track.enabled, mid: this.getSendTrackMid(track) })
+      }
+    });
+    var buttonText = document.getElementById("mute-microphone")?.innerText
+    if (buttonText == "Mute microphone") {
+      document.getElementById("mute-microphone")!.innerText = "Unmute microphone";
```

Comment:
You query the same elements twice. Do it once, query button and then you can check for it's innerText and then change it.